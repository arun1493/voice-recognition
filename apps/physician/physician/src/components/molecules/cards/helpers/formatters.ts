import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
