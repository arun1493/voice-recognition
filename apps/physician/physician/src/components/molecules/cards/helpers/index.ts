export * from "./labels";
export * from "./formatters";
