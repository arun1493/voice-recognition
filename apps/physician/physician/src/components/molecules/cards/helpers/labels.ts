import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
