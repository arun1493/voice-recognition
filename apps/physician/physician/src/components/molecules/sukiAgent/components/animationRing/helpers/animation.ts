import { RING_ANIMATION_DURATION, RING_BASE_OPACITY, RING_SCALE, SukiAgentState } from "../../../../../../constants";

function isValidSukiAgentStateForSingleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (isAgentActive && agentStatus === SukiAgentState.AWAKE) || agentStatus === SukiAgentState.PROCESSING;
}

function isValidSukiAgentStateForMultipleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (
    isAgentActive &&
    !isValidSukiAgentStateForSingleRingAnimation(agentStatus, isAgentActive) &&
    agentStatus === SukiAgentState.LISTENING
  );
}

function getDictationRingVariants(ringsCount: number) {
  const dictationRingVariants = [];
  for (let i = ringsCount; i > 0; i--) {
    const RING_VARIANTS = {
      oscillate: {
        opacity: RING_BASE_OPACITY - i * 0.2,
        scale: RING_SCALE + i * 0.01,
        transition: {
          yoyo: Infinity,
          duration: RING_ANIMATION_DURATION,
          type: "tween",
          delay: i * 0.1,
          ease: "easeInOut"
        }
      }
    };

    dictationRingVariants.push(RING_VARIANTS);
  }
  return dictationRingVariants;
}

export {
  getDictationRingVariants,
  isValidSukiAgentStateForMultipleRingAnimation,
  isValidSukiAgentStateForSingleRingAnimation
};
