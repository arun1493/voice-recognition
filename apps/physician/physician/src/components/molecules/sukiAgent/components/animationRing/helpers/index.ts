export * from "./ringSize";
export * from "./animation";
