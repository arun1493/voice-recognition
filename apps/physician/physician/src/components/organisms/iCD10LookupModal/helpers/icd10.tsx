import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
