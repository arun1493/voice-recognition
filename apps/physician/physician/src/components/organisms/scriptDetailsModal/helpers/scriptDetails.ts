import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
