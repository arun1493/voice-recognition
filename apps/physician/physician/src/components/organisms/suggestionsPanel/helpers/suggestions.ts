import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
