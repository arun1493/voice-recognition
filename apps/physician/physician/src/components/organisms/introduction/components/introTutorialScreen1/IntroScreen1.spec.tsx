import "@testing-library/jest-dom/extend-expect";
import { cleanup, RenderResult } from "@testing-library/react";

import { IntroScreen1 } from ".";
import { componentSetup } from "../../../../../hocs";
import { configureI18n } from "../../../../../helpers";
import { shouldMatchSnapshot } from "../../../../../helpers/snapshot.spec";

const setup = componentSetup(IntroScreen1);

describe("Intro Screen1", () => {
  let component: RenderResult;

  beforeEach(() => {
    configureI18n();
    component = setup();
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    const { baseElement } = component;
    expect(baseElement).toBeTruthy();
  });

  it("should render the correct header", () => {
    const { baseElement } = component;
    expect(
      baseElement.getElementsByTagName("h1")?.[0]?.innerHTML ||
        baseElement.getElementsByTagName("h3")?.[0]?.innerHTML ||
        baseElement.getElementsByTagName("h4")?.[0]?.innerHTML
    ).toEqual("Tap here to wake Suki");
  });

  it("should render the correct description", () => {
    const { baseElement } = component;
    expect(baseElement.getElementsByTagName("p")[0].innerHTML).toEqual("Suki only listens when awake.");
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(IntroScreen1);
  });
});
