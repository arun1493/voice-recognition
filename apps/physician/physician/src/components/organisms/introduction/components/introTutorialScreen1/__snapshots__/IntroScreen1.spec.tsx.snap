// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro Screen1 should match snapshot 1`] = `
<div
  className="introScreen1styled__StyledIntroContainer-okg1hb-1 hPMcsv"
>
  <div
    className="intro-screen1-header-container"
  >
    <h1
      className="headerstyled__StyledH1-sc-1y4xvz5-0 ifdpFf intro-screen1-header"
      data-cy="intro-screen1-header"
    >
      Tap here to wake Suki
    </h1>
    <svg
      fill="none"
      height="100"
      viewBox="0 0 64 100"
      width="64"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clipPath="url(#clip0_1336_785)"
      >
        <path
          d="M9.52685 65.2369H18.7602C19.7739 65.2369 20.63 64.4789 20.7285 63.47C21.2161 58.4752 21.2809 53.2094 20.6391 47.2016C18.1517 23.9211 6.03453 7.3684 1.15706 1.2466C0.679345 0.64701 1.38043 -0.26838 2.0677 0.0747866C14.8499 7.02028 40.9238 29.2761 44.7115 63.4039C44.8259 64.4355 45.6891 65.2369 46.727 65.2369H55.6937C57.3605 65.2369 58.296 67.156 57.2692 68.469L35.7612 95.971C34.1597 98.0188 31.0609 98.0188 29.4594 95.971L7.95141 68.469C6.92463 67.156 7.86009 65.2369 9.52685 65.2369Z"
          fill="url(#paint0_linear_1336_785)"
        />
      </g>
      <defs>
        <linearGradient
          gradientUnits="userSpaceOnUse"
          id="paint0_linear_1336_785"
          x1="8.13089"
          x2="34.4261"
          y1="3.56545"
          y2="63.7324"
        >
          <stop
            stopColor="#F0F4EF"
          />
          <stop
            offset="1"
            stopColor="#CCCCCC"
          />
        </linearGradient>
        <clipPath
          id="clip0_1336_785"
        >
          <rect
            fill="white"
            height="100"
            width="64"
          />
        </clipPath>
      </defs>
    </svg>
  </div>
  <div
    className="introScreen1styled__SukiAgentSpacer-okg1hb-0 eHqHDs"
  />
  <p
    className="textstyled__StyledText-sc-1vedkxi-0 kTwEkJ"
    color="inherit"
    cursor="default"
    data-cy="intro-screen1-description"
    onClick={[Function]}
    size="xl"
  >
    Suki only listens when awake.
  </p>
</div>
`;
